src/npc.cpp

```diff
@@ -285,18 +285,19 @@ void Npc::onRemoveCreature(Creature* creature, bool isLogout)
 	}
 }
 
-void Npc::onCreatureMove(Creature* creature, const Tile* newTile, const Position& newPos, const Tile* oldTile,
-                         const Position& oldPos, bool teleport)
+void Npc::onCreatureMove(std::shared_ptr<Creature> creature, std::shared_ptr<const Tile> newTile,
+                         const Position& newPos, std::shared_ptr<const Tile> oldTile, const Position& oldPos,
+                         bool teleport)
 {
 	Creature::onCreatureMove(creature, newTile, newPos, oldTile, oldPos, teleport);
 
-	if (creature == this || creature->getPlayer()) {
+	if (creature.get() == this || creature->getPlayer()) {
```

Comment:
We’ve got some legacy code here, hehe!
```suggestion
	if (creature == getNpc() || creature->getPlayer()) {
```